### DISCUSSION TITLE
Reserve VM id

### DISCUSSION THREAD

**Discussion by Kukkimonsuta** [bpg/terraform-provider-proxmox#1456] [Q&A] [answered]
Is it somehow possible to create blank VM, get the auto generated VM ID, create other resources based on that and then finish creating the VM? Specifically what I'd like to do is have cloud config file per VM otherwise I get warnings like `Warning: the existing file "local:snippets/cloud-config.yaml" has been overwritten by the resource` and using vm_id would be probably the best as they can't change and it would be consistent with disk ids.

Simplified example:
```tf
resource "proxmox_virtual_environment_file" "cloud_config" {
  content_type = "snippets"
  datastore_id = "local"
  node_name = var.target_node

  source_raw {
    data = <<-EOF
    #cloud-config
    ...snip...
    EOF

    file_name = "${proxmox_virtual_environment_vm.vm.vm_id}-cloud-config.yaml"
  }
}

resource "proxmox_virtual_environment_vm" "vm" {
  # .. snip ..

  initialization {
    user_data_file_id = proxmox_virtual_environment_file.cloud_config.id
  }

  # .. snip ..
}
```

**Comment by bpg-dev** [comment, marked as answer]
The "blank VMs" won't be possible. But you can manually allocate VM IDs and pass them to both the VM (set as [`vm_id`](https://registry.terraform.io/providers/bpg/proxmox/latest/docs/resources/virtual_environment_vm#vm_id) attribute) and the file resource. 
You can probably use [`count`](https://developer.hashicorp.com/terraform/language/meta-arguments/count) or define a map / array to manage your VM IDs
